Add links to generate a mail template to subscribe and unsubscribe for more convinience

daipom opened the pull request:
Fix mail addresses into mailto tags so that we can subscribe the address
just by clicking the link and sending the mail, which is automatically created.

I haven't finished this yet.

I will run the rakefile, and check if this is working properly.
I need to add the diff of `index.html` after running the rakefile.

daipom committed:
Fix mail addresses into mailto tags

piroor commented:
差分の方は特に問題はないように感じましたが、コミットメッセージ（とプルリクエストのタイトル）としては
Generate links to start message composition with template to subscribe for more convenience
とかの方が説明的でよいのではないだろうか？と思いました。
また、リンクのラベルをSubscribeとして、同様にUnsubscribeというラベルでunsubscribe用のリンクも用意するといいのではないかと思いました。
いかがでしょう？

kou commented on Rakefile:
```diff
@@ -80,7 +80,7 @@ class RepositoryListUpdater
               <td><a href="#{owner_url(fqdn, owner)}">#{h(owner)}</a></td>
               <td>#{repository_column}</td>
-              <td>#{h(to)}</td>
+              <td><a href="mailto:#{h(to)}?cc=null@commit-email.info&amp;subject=Subscribe&amp;body=subscribe">#{h(to)}</a></td>
```
Could you extract this `mailto:` URL generate code as a method like `owner_url` and `repository_url`?

daipom replied:
Thank you for your advise.
It will make the code very readable.
I will fix it.

daipom replied:
I have fixed this.

daipom commented:
@piroor ありがとうございます。
メッセージも、リンクのことも、その方が分かりやすくて良いと思います。
タイトル変更して、コードも修正してみます。

daipom committed:
Add subscribe and unsubsribe columns to get mail template

Also add some text explaining this to "How to use" section.

daipom commented:
<img src="https://user-images.githubusercontent.com/12229857/148891086-002c526c-d825-4722-a88b-7a1934021375.png" width="500px">

daipom commented:
I have added `Subscribe` and `Unsubscribe` columns to `Repositories` section.
We can get the mail template by clicking these buttons and can send it as it is to subscribe or subscribe.

I have also added some texts explaining this to `How to use` section.

kou commented on Rakefile:
```diff
@@ -92,6 +96,14 @@ class RepositoryListUpdater
     "#{owner_url(fqdn, owner)}#{h(repository)}/"
   end
+
+  def mailto_href_subscribe(to)
```
`href` is for `<a>`. This is used for `href` but this is an URL not `href`.

```suggestion
  def subscribe_url(to)
```

(I think that `mailto` is redundant here. Should we include it in method name?)

daipom replied:
Thank you for the advice.
That's exactly right.
I have fixed this.

kou commented on ansible/files/var/www/html/index.html:
```diff
@@ -44,6 +44,8 @@ <h3>How to subscribe</h3>
 subscribe
         </pre>
+
+        <p>* You can get the mail template by clicking the "Subscribe" button in the list and can send it as it is.</p>
```
```suggestion
        <p>* You can get the mail template by clicking the "Subscribe" link in the list and can send it as it is.</p>
```

daipom replied:
Thank you for the suggestion.
I have fixed this.

kou commented on Rakefile:
```diff
@@ -92,6 +96,14 @@ class RepositoryListUpdater
   end
+
+  def mailto_href_subscribe(to)
+    "mailto:#{h(to)}?cc=null@commit-email.info&amp;subject=Subscribe&amp;body=subscribe"
```
Ah, we should call `h` when we use an URL not generate an URL. `&amp;` is for HTML not for URL.
I've updated the current implementation. Please follow the convention.

daipom replied:
Thank you for the advice.
We should enclose the whole URL in `h` and leave the conversion to `&amp;` to `h`, right?

```rb
h("mailto:#{to}?cc=null@commit-email.info&subject=Subscribe&body=subscribe")
```

daipom replied:
Oh, I should refer to this fix. Thank you.
https://github.com/kou/commit-email.info/commit/38c11ffc8f244e921c52a1ad0320693094210294

kou replied:
This method should not call `h`.
See also: https://github.com/kou/commit-email.info/commit/38c11ffc8f244e921c52a1ad0320693094210294

daipom replied:
Thank you. I understand.
`h` must be called on href directly.
I have fixed this.

daipom committed:
Fix some method names

because the names are redundant and these are not `href` but `URL`.

daipom committed:
Fix `button` to `link`

daipom committed:
Use h() on href not URL generation about subscribing links

daipom commented:
I haven't been able to confirm the behavior with vagrant,
but I have confirmed that opening the html file in a browser and clicking on the link will open the intended email.

kou commented:
Thanks!

daipom commented:
@kou @piroor 
Thank you very much for reviewing and merging!
